### PR TITLE
Merge LuckyMigrator with LuckyRecord

### PR DESCRIPTION
**Pull request by paulcsmith**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-          command: shards install
-      - run:
-          name: "Check formatting"
-          command: crystal tool format src spec --check
-      - run:
-          name: "Migrate database"
-          command: crystal tasks.cr -- db.migrate
-      - run:
-          name: "Running specs"
-          command: crystal spec
```
Comment (paulcsmith) [inline .circleci/config.yml]: Switching to Travis :D

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /docs/
 /lib/
-/bin/
+/bin/lucky
+/bin/gen.migration*
```
Comment (paulcsmith) [inline .gitignore]: Don't commit generated binaries (precompiled tasks like `gen.migration`)

File: .travis.yml [hunk 3]
```diff
@@ -1 +1,22 @@
...
+  - shards install
+matrix:
+  allow_failures:
+    - crystal: nightly
+script:
+  - crystal tasks.cr -- db.migrate
+  - crystal tasks.cr -- db.rollback_all
+  - crystal tasks.cr -- db.migrate.one
+  - crystal tasks.cr -- db.migrate
+  - crystal tasks.cr -- gen.migration TestMigration
```
Comment (paulcsmith) [inline .travis.yml]: Test that all tasks work and migrations run

File: db/migrations/20170127143149_create_users.cr [hunk 4]
```diff
@@ -1,4 +1,4 @@
-class CreateUsers::V20170127143149 < LuckyMigrator::Migration::V1
+class CreateUsers::V20170127143149 < LuckyRecord::Migrator::Migration::V1
```
[the remark this answers is not in the document]
Comment (edwardloveall) [reply db/migrations/20170127143149_create_users.cr]: Yeah reading through more of the code I think this makes the most sense

File: .travis.yml [hunk 5]
```diff
@@ -1 +1,22 @@
 language: crystal
+crystal:
+  - latest
+  - nightly
```
Comment (edwardloveall) [inline .travis.yml]: Do we maybe want to also set the version explicitly?

File: db/migrations/20170127143149_create_users.cr (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulcsmith) [reply db/migrations/20170127143149_create_users.cr]: Yeah exactly. It was to keep it nicely namespaces. Also, you typically use a generator for these so I think the extra characters are ok

File: shard.yml [hunk 6]
```diff
@@ -3,6 +3,10 @@ version: 0.6.0
 
 crystal: 0.25.0
 
+targets:
+  gen.migration:
+    main: src/precompiled_tasks/gen/migration.cr
```
[the remark this answers is not in the document]
Comment (paulcsmith) [reply shard.yml]: :D

File: .travis.yml (hunk 5 above)
Comment (paulcsmith) [reply .travis.yml]: I wish you could, but I could not figure out a way to do it :(

**Comment by paulcsmith** [comment]
Thanks for taking a look! I'll squash and re-run specs to make sure everything is ok

**Pull request merged**